src/app/content/utils/seoUtils.ts

```diff
@@ -9,20 +14,101 @@ import {
   splitTitleParts,
 } from './archiveTreeUtils';
 
-const getParentPrefix = (node: LinkedArchiveTreeNode | undefined): string => {
+const domParser = new DOMParser();
+
+export const getParentPrefix = (node: LinkedArchiveTreeNode | undefined, includeTitle: boolean = false): string => {
   if (!node) {
     return '';
   }
 
   if (archiveTreeSectionIsChapter(node)) {
-    const number = getArchiveTreeSectionNumber(node);
-    return `Ch. ${number} `;
+    const number = getArchiveTreeSectionNumber(node).trim();
+    const name = getArchiveTreeSectionTitle(node);
+    return includeTitle ? `Ch. ${number}. ${name}` : `Ch. ${number} `;
   }
 
   return archiveTreeSectionIsBook(node.parent)
     ? getArchiveTreeSectionTitle(node) + ' '
-    : getParentPrefix(node.parent);
+    : getParentPrefix(node.parent, includeTitle);
+
+};
+
+const hideMath = (node: Element) => {
+  node.querySelectorAll('math, .os-math-in-para').forEach((el: Element) => {
+    el.outerHTML = '...';
+  });
+  return node;
+};
+
+export const getTextContent = (str: string) => {
+  const parsed = domParser.parseFromString(str, 'text/html');
+  const text = parsed.body.textContent;
+  return text;
+};
+
+const removeExcludedContent = (node: Element) => {
+  if (!node) {
+    return null;
+  }
+  const excludedContent = node.querySelectorAll(
+    '[data-type="abstract"], .learning-objectives, .chapter-objectives, .be-prepared, .os-teacher'
+  ) || [];
+
+  for (const item of Array.from(excludedContent)) {
```

Comment:
`Array.from` is not needed

Reply:
Without it I get a TS error: `Type 'NodeListOf<Element>' must have a '[Symbol.iterator]()' method that returns an iterator.`

Reply:
thats weird.... mdn is pretty sure `for...of` loops should work for nodelists

Reply:
I assume it may be related to ts version

Reply:
Most solutions I've found recommend setting es6 as the "target" or as a "lib" in tsconfig, but so far that's not working for me. Fwiw it seems like `Array.from` is used across the project where we iterate over or destructure nodelists.

Reply:
Yea, I think it is not a big deal if it is not working by default yet